src/snmalloc/ds_core/bits.h

```diff
@@ -203,7 +207,12 @@ namespace snmalloc
       overflow = __builtin_mul_overflow(x, y, &prod);
       return prod;
 #elif defined(_MSC_VER)
-#  ifdef _WIN64
+#  if defined(_M_ARM64) || defined(_M_ARM64EC)
+      size_t high_prod = __umulh(x, y);
+      size_t prod = x * y;
+      overflow = high_prod != 0;
+      return prod;
+#  elif defined(_WIN64)
```

Comment:
Would it be better to use the same code with `__umulh` for the x64 version as well?  It seems that would be optimised to the same code, and then we have slightly less branching in the `ifdef`s.

Reply:
Sure, I'll take a look.

Reply:
@mjp41, on X64 the code generated for the `__umulh` version contains two multiplications, whereas the one for `_umul128` contains only one. Would you like me to use `__umulh` in both cases or keep it as is?

```
_umul128
00007FF663A51000  mov         r9,rdx  
00007FF663A51003  mov         rax,rdx  
00007FF663A51006  mul         rax,rcx  
00007FF663A51009  test        rdx,rdx  
00007FF663A5100C  setne       al  
00007FF663A5100F  imul        rcx,r9  
00007FF663A51013  mov         byte ptr [r8],al  
00007FF663A51016  mov         rax,rcx  

__umulh
00007FF663A51020  mov         rax,rdx  
00007FF663A51023  mul         rax,rcx  
00007FF663A51026  test        rdx,rdx  
00007FF663A51029  setne       cl  
00007FF663A5102C  mov         byte ptr [r8],cl  
00007FF663A5102F  ret  
```

Reply:
Thanks. Let's leave it as is. That is on the fast path for `calloc`, so code gen is important.